25_28_10/src/functions.js

```diff
@@ -54,3 +54,52 @@ function appendSelectHtml(callback, options, parentEl, defaultOptionName) {
 
   return selectEl;
 }
+
+const addCarSelect = (cars) => {
+  usersBlock.childNodes.forEach((element) => {
+    let markup = `<select class='selectJS'><option>${cars[0]?.brand}</option><option>${cars[1]?.brand}</option><option>${cars[2]?.brand}</option><option>${cars[3]?.brand}</option></select>`;
+    element.insertAdjacentHTML("beforeend", markup);
+  });
+};
+
+const addSelectedCar = (event) => {
+  const cardBody = event.target.closest("div");
+  const carName = event.target.value;
+  const markup = `<span>${carName}</span>`;
+  cardBody.insertAdjacentHTML("beforeend", markup);
+
+  const currentUser = cardBody.querySelector(".nameJS");
+  const userName = currentUser.innerHTML;
+  let currentUserObj;
+
+  users.filter((e) => {
+    if (userName === e.name) {
+      currentUserObj = e;
+    }
+  });
```

Comment:
```suggestion
  const currentUserObj = users.find(({ name }) => name === userName);
```

а вообще лучше было бы искать по айдишнику.